rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -208,12 +208,47 @@ public void setPendingFederation(PendingFederation federation) {
         pendingFederation = federation;
     }
 
+    @Override
+    public void clearPendingFederation() {
+        setPendingFederation(null);
+    }
+
+    @Override
+    public Optional<Federation> getProposedFederation(FederationConstants federationConstants, ActivationConfig.ForBlock activations) {
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (proposedFederation != null || isProposedFederationSet) {
+            return Optional.ofNullable(proposedFederation);
+        }
+
+        Optional<Integer> storageVersion = getStorageVersion(PROPOSED_FEDERATION_FORMAT_VERSION.getKey());
```

Comment:
And if `storageVersion` is null? Consider this when writing the tests

Reply:
done